client/json-rpc/rpc_procedures.c

```diff
@@ -549,6 +553,17 @@ cJSON * method_xdag_do_xfer(struct xdag_rpc_context * ctx, cJSON *params, cJSON
 			if (cJSON_IsString(json_address)) {
 				strcpy(address, json_address->valuestring);
 			}
+
+			cJSON *json_remark = cJSON_GetObjectItem(param, "remark");
+			if (cJSON_IsString(json_remark)) {
+				if(strlen(json_remark->valuestring) < 32) {
```

Comment:
strlen is usually unsafe, where it is checked that there is \0 somewhere?
when this is false -> if(strlen(json_remark->valuestring) < 32)  it means that there isn't the \0 so it is already on buffer overflow and likely crashed, so i don't understand well why you check if there is an error in this way.

> SYNOPSIS        
>        #include <string.h>
> 
>        size_t strnlen(const char *s, size_t maxlen);

You should be able to add max size to strlen to make it safe.